WordPress/gutenberg
# [Site Editor]: Fix multiple templates creation while in the process of creating one

## What?
Mentioned by @annezazu here: https://github.com/WordPress/gutenberg/issues/43877#issuecomment-1246070721

Currently when we're in the process of creating a template in Site Editor, a user can trigger the creation flow for multiple templates. This PR fixes that by adding an overlay with a spinner during the request to create a template.


Based on the below  @jameskoster 's https://github.com/WordPress/gutenberg/pull/44146#issuecomment-1246437841







## Testing Instructions
1. In site editor you should be able to create a single template at the time, no matter how many times you click a menu item or button to create one.
2. The issues is more visible on slow networks.

### Before

https://user-images.githubusercontent.com/26996883/190030104-dacafbf2-3d90-46fe-9120-0f5ff2ddbc8a.mov

### After
I'm not creating the actual templates in the below video to showcase the overlay.


https://user-images.githubusercontent.com/16275880/190187590-52e3f3e7-c5f8-45f0-9fb0-e8b8f06c1b64.mov
